hw_diag/utilities/quectel.py

```diff
@@ -289,64 +338,68 @@ def ensure_modem_manager_health() -> None:
     try:
         all_modems = mm_proxy.get_all_modems()
     except Exception as e:
-        logging.error(f'failed to get modem list with error: {e}')
-        logging.error('most likely modem manager dbus interface is not responsive')
+        logging.error(f"failed to get modem list with error: {e}")
+        logging.error("most likely modem manager dbus interface is not responsive")
 
-    if (len(all_modems) == 0):
+    if len(all_modems) == 0:
         reset_modem_manager()
 
 
 def is_att_sim() -> bool:
     att_sim = False
     try:
         modem = find_eg25g_modem()
-        sim = modem.get_sim()
-        att_sim = sim.is_att_sim()
+        if modem:
+            sim = modem.get_sim()
+            att_sim = sim.is_att_sim()
     except Exception as e:
         # no modem or no sim
-        logging.info(f'failed to get sim info with error: {e}')
+        logging.info(f"failed to get sim info with error: {e}")
     return att_sim
 
 
 def ensure_quectel_health() -> None:
-    '''
+    """
     ensure that modem manager is replying over dbus.
     download firmware files based on modem's hw revision.
     update modem's settings to Data Centric mode.
     if modem has AT&T SIM, update modem's firmware
     we are limiting firmware upgrade to AT&T SIM for now because we have seen
     in testing that firmware update can effect non at&t connectivity like T-Mobile
     only gets ipv6 after firmware update which can create problems for us.
-    '''
+    """
     # make sure modem manager is responsive
     ensure_modem_manager_health()
 
     # we want to download firmware files at the first possible opportunity.
     try:
```

Comment:
Isn't it a waste to download the firmware if not on ATT?

Reply:
We download and keep it as the plan is to upgrade firmware in all cases down the line. I also wanted to cover the case where a AT&T sim is installed in field.

Reply:
Yea I guess that is the most user friendly behavior. Is the download at least only happening if a modem is detected?